GenumKit/Parsers/FontsFileParser.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  FontsFileParser.swift
+//  Pods
+//
+//  Created by Derek Ostrander on 3/7/16.
+//
+//
+
+import Foundation
+import AppKit.NSFont
+
+typealias Font = (name: String, variant: String)
+
+extension CFString {
+    var alphanumericString: String {
+         return (self as String).componentsSeparatedByCharactersInSet(NSCharacterSet.alphanumericCharacterSet().invertedSet).joinWithSeparator("")
+    }
+}
+
+extension CTFont {
+    static func loadFont(path: String) -> Font? {
+        let inData = NSData(contentsOfFile: path)
+        var error: Unmanaged<CFError>?
+        guard
+            let provider: CGDataProvider = CGDataProviderCreateWithCFData(inData),
+            let font = CGFontCreateWithDataProvider(provider)
+            where CTFontManagerRegisterGraphicsFont(font, &error) else {
+                return nil
+        }
+        return CTFontCreateWithGraphicsFont(font, 0, nil, nil).font
+    }
+
+    private var font: Font? {
+        let family = CTFontCopyFamilyName(self).alphanumericString
+        let name = CTFontCopyFullName(self).alphanumericString.stringByReplacingOccurrencesOfString(family, withString: "")
+        if family.characters.count == 0 {
+            print("There was a problem detecting the name of the font. \(CTFontCopyFamilyName(self)) -- \(CTFontCopyFullName(self)) ")
+            return nil
+        }
+        return (family, name.characters.count > 0 ? name : "Regular")
+    }
+}
+
+// MARK: Interface
+public final class FontsFileParser {
+    public var entries: [String: Set<String>] = [:]
+
+    public init() {}
+
+    public func parseFonts(path: String) {
+        for file in contentsOfDirectory(path) {
+            let fullPath = (path as NSString).stringByAppendingPathComponent(file)
```

Comment:
TBH, I used `mdfind /Library/Fonts/Arial.ttf` to display the metadata of a random font file which gave me:
```
kMDItemContentTypeTree         = (
    "public.truetype-ttf-font",
    "public.truetype-font",
    "public.font",
    "public.data",
    "public.item"
)
```
So that's how I saw the UTI tree of the font file and guessed that the generic UTI for any font file is `public.font`